test/jdk/java/security/ProtectionDomain/RecursionDebug.java

```diff
@@ -87,7 +87,5 @@ public static void main(String[] args) throws Exception {
             throw new Exception
                 ("Test with custom non-bootclass SecurityManager failed");
         }
-
-        System.setSecurityManager(null);
```

Comment:
Why did this line need to be removed?

Reply:
This is where the `setSecurityManager` method is called again when a security manager has already been installed, and the `getProtectionDomain` permission is needed. While I can add the new permission into the policy file, my understanding is that this line was just a clean-up and the test was about implementation of `ProtectionDomain::toString` (esp the `seeAllp` method called by it) when debug is on.

Reply:
Ok, thanks for the explanation. I think it should be ok to remove this line then.